astrosat/management/commands/unloaddata.py

```diff
@@ -0,0 +1,180 @@
+        if has_lzma:
+            self.compression_formats["lzma"] = (lzma.LZMAFile, "r")
+            self.compression_formats["xz"] = (lzma.LZMAFile, "r")
+
+        for fixture_label in fixture_labels:
+            if self.find_fixtures(fixture_label):
+                break
+        else:
+            return
+
+        # here is the different bit...
+        # just execute the single unload_label fn, instead of all the
+        # trial-and-error that happens when actually loading a fixture
+        self.class_deletion_record = collections.defaultdict(int)
+        for fixture_label in fixture_labels:
+            self.unload_label(fixture_label)
+
+        if self.verbosity >= 1:
+            self.stdout.write(
+                "Unloaded %d object(s) (of %d) from %d fixture(s)"
+                % (
+                    self.unloaded_object_count,
+                    self.fixture_object_count,
+                    self.fixture_count,
+                )
+            )
+
+    def unload_label(self, fixture_label):
+
+        show_progress = self.verbosity >= 3
+        for fixture_file, fixture_dir, fixture_name in self.find_fixtures(fixture_label):
+            _, ser_fmt, cmp_fmt = self.parse_name(os.path.basename(fixture_file))
+            open_method, mode = self.compression_formats[cmp_fmt]
+            fixture = open_method(fixture_file, mode)
+            try:
+                self.fixture_count += 1
+                objects_in_fixture = 0
+                unloaded_objects_in_fixture = 0
+                if self.verbosity >= 2:
+                    self.stdout.write(
+                        "Installing %s fixture '%s' from %s."
+                        % (ser_fmt, fixture_name, humanize(fixture_dir))
+                    )
+
+                objects = serializers.deserialize(
+                    ser_fmt,
+                    fixture,
+                    using=self.using,
+                    ignorenonexistent=self.ignore,
+                    handle_forward_references=True,
+                )
+
+                for obj in objects:
+                    objects_in_fixture += 1
+                    if (
+                        obj.object._meta.app_config in self.excluded_apps
+                        or type(obj.object) in self.excluded_models
+                    ):
+                        continue
+
+                    # here is the different bit...
+                    obj_class_label = obj.object._meta.label
+                    try:
+                        n_deleted, deleted_classes = obj.object.delete(using=self.using)
+                        unloaded_objects_in_fixture += n_deleted
+                        for key, val in deleted_classes.items():
+                            if key == obj_class_label:
+                                self.class_deletion_record[key] += (val - 1)
+                            else:
+                                self.class_deletion_record[key] += val
```

Comment:
More for my learning but would you recommend doing this as a **ternary if**? e.g.

`self.class_deletion_record[key] += (val - 1) if key == obj_class_label else self.class_deletion_record[key] += val`. I would do it this way normally in Java or JavaScript, not so sure here, not sure it is more readable.

Reply:
I guess I could.  It just seems less readable in this instance.

Reply:
I guess I could.  It just seems less readable in this instance.